File: backend/app/models/task.py
from sqlalchemy import Column, Integer, String, Text, DateTime, Enum
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.sql import func
import enum

Base = declarative_base()

class TaskStatus(enum.Enum):
    IN_PROGRESS = "inprogress" 
    COMPLETED = "completed"
    ARCHIVED = "archived"

class TaskPriority(enum.Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"

class Task(Base):
    __tablename__ = "tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    title = Column(String(255), nullable=False, index=True)
    description = Column(Text, nullable=True, index=True)
    status = Column(Enum(TaskStatus), default=TaskStatus.IN_PROGRESS)
    priority = Column(Enum(TaskPriority), default=TaskPriority.MEDIUM)
    due_date = Column(DateTime, nullable=True, index=True)
    created_at = Column(DateTime, default=func.now())
    updated_at = Column(DateTime, default=func.now(), onupdate=func.now())

File: client/src/types/task.ts
export interface Task {
  id: number;
  title: string;
  description?: string;
  status: 'inprogress' | 'completed' | 'overdue' | 'archived';
  priority: 'LOW' | 'MEDIUM' | 'HIGH';
  due_date?: string;
  created_at: string;
  updated_at: string;
}

export interface ChatMessage {
  type: 'user' | 'agent' | 'system';
  message: string;
  timestamp: Date;
}
